### DISCUSSION TITLE
Waveforms fix for Mixxx on Raspbian OS

### DISCUSSION THREAD

**Discussion by neoben** [fayaaz/mixxx-pi-gen#74] [General]
Hi @fayaaz,
this is Carmine and I first of all I would like to say that this is a great project!

I'm wondering if you can give me some insight about how to fix the waveforms on Mixxx running on Rasperry Pi.
I have a Rasberry PI 5 and I'm running the latest Raspian OS.
I have installed Mixxx from the official apt repository and I have issue with waveforms.
Since the waveforms are working fine on this project can you advice how to fix them?
Is it possibile to fix the issue installing additional software or changing system configuration without patching the code?

Thanks a lot!

**Comment by fayaaz** [comment]
I can't tell you why the waveforms don't work in the debian version, I never investigated why, however you can build Mixxx 2.4 from source.

Here's where I do it as part of the image: 
https://github.com/fayaaz/mixxx-pi-gen/blob/main/stage3/01-install-packages/01-run.sh

You could also get the  .deb from the [release](https://github.com/fayaaz/mixxx-pi-gen/releases/download/v0.9.1/mixxx-2.4-beta-328-gf610d58a25.deb) or from actions. You can install this with `sudo apt install ./mixxx-2.4-beta-328-gf610d58a25.deb` after downloading it.

**Reply by neoben** [reply]
Thanks a lot!
So I suppose that Mixxx 2.4 includes the fix for waveforms.
Thanks,

Carmine

**Reply by fayaaz** [reply]
I don't know - you could try building Mixxx 2.3 on the raspberry pi and the waveforms might work. Debian heavily patch the software in their repos. Iirc I wanted to compile Mixxx on a pi2 without the debian patches for performance reasons and it took a good few hours.
